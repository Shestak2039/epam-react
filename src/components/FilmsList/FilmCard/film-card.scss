@import "../../../styles/colors";

.film-wrapper {
  cursor: pointer;
  transition: all 0.5s;

  position: relative;

  &:hover {
    transition: all 0.5s;

    .additional-info-icon {
      position: absolute;
      right: 10px;
      top: 10px;

      width: 30px;
      height: 30px;

      color: $default-text-color;
      background-color: $main-theme;
      border-radius: 100%;

      &:after {
        content: '\2807';

        position: absolute;
        left: 10px;
        top: 1px;

        font-size: 20px;
      }
    }
  }

  .film-info {
    display: flex;
    justify-content: space-between;

    margin-top: 10px;

    color: $default-text-color;

    &-left {
      &-title {
        font-weight: bold;
      }

      &-genres {
        font-size: 14px;
        color: $pink-text-color;

        margin-top: 10px;
      }
    }

    &-right {
      &-year {
        font-size: 15px;

        border: 1px solid $secondary-color;

        padding: 5px;
      }
    }
  }
}
