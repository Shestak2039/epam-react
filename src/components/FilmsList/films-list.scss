@import "../../styles/colors";

.films-wrapper {
  padding-top: 15px;

  .count-movies {
    font-size: 18px;
    color: $default-text-color;
  }

  .films {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
    grid-gap: 20px;

    margin-top: 15px;
  }
}
