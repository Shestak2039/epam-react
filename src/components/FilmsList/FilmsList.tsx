import React from 'react';

import './films-list.scss';
import FilmCard, { FilmCardModel } from './FilmCard/FilmCard';

export interface FilmsListProps {
    films: FilmCardModel[];
    openPage: (id: number) => void;
}

const FilmsList: React.FunctionComponent<FilmsListProps> = ({films, openPage}) => {
    return (
        <div className="films-wrapper">
            <div className="count-movies">{films.length} movies found</div>
            <div className="films">
                {films.map(film => <FilmCard key={film.id} film={film} openPage={openPage}/>)}
            </div>
        </div>
    )
};

export default FilmsList;
