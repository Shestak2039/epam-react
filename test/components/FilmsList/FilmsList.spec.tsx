import React from 'react';
import { render } from '@testing-library/react';

import FilmsList from '../../../src/components/FilmsList/FilmsList';
import { FilmCardModel } from '../../../src/models/film-card.model';
import { BrowserRouter } from 'react-router-dom';

describe('FilmsList', () => {
    const testMovie1: FilmCardModel = {
        id: 'id',
        title: 'title',
        genres: [],
        year: 'year',
        imageUrl: 'imageUrl',
        rating: 0,
        durationInMinutes: 0,
        overview: 'overview',
    };
    
    test('renders FilmsList snapshot', () => {
        const { asFragment } = render(<BrowserRouter><FilmsList films={[testMovie1]}/></BrowserRouter>);

        expect(asFragment()).toMatchSnapshot();
    });
});
