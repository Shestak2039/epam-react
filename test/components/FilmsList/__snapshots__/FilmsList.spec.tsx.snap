// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList renders FilmsList snapshot 1`] = `
<DocumentFragment>
  <div
    class="films-wrapper"
  >
    <div
      class="count-movies"
    >
      1 movies found
    </div>
    <div
      class="films"
    >
      <div
        class="film-wrapper"
      >
        <a
          href="/film/id"
        >
          <img
            height="450"
            src="imageUrl"
            width="300"
          />
        </a>
        <div
          class="film-info"
        >
          <div
            class="film-info-left"
          >
            <div
              class="film-info-left-title"
            >
              title
            </div>
            <div
              class="film-info-left-genres"
            />
          </div>
          <div
            class="film-info-right"
          >
            <div
              class="film-info-right-year"
            >
              year
            </div>
          </div>
        </div>
        <div
          class="additional-info-icon"
          data-testid="additional-icon"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
